server.js

```diff
@@ -2,9 +2,20 @@ const net = require('net');
 const JsonSocket = require('json-socket');
 
 
-// Object containing clients list and relative information as such
-// {clientId: {socket: ..., charactersMaxLength: ...}, ...}
-const clients = {};
+const MAX_CONCURRENT_JOBS = 50;
```

Comment:
For this you need to implement a scheduler on top of the message queue, what you do it is not very clear and if the server cashes you loose all the waiting requests. @triglian add a reference you trust as guideline to implement a scheduler